girder/girder
# Python coverage cleanup in CircleCI

* Attempt to submit pytest results to CircleCI for analysis
* Remove the PYTHON_COVERAGE CMake option
  * Python coverage will now always run. It was already always being run for Pytest tests.
* Remove the PYTHON_BRANCH_COVERAGE CMake option
  * This option should be set directly the Coverage config file (which is the only way to set it for Pytest).
* Remove the PYTHON_COVERAGE_CONFIG CMake option and unify Coverage configs
  * Both the legacy CMake+unittest and Pytest runners will now use the same .coveragerc file.
* Always write coverage reports to "build/test/coverage"
  * This ensures that Pytest-generated coverage reports are written to the same place as unittest / CMake-generated reports.
* Write intermediate Python coverage files inside "build/test/coverage"
* Reorganize the root .gitignore file                                                                                                                                                      
  * This also stops ignoring the "htmlcov" directory, which should not longer be generated by Javascript coverage.
